contribs/gmf/src/directives/editfeature.js

```diff
@@ -0,0 +1,446 @@
+goog.provide('gmf.EditfeatureController');
+goog.provide('gmf.editfeatureDirective');
+
+
+goog.require('gmf');
+goog.require('gmf.EditFeature');
+goog.require('gmf.XSDAttributes');
+/** @suppress {extraRequire} */
+goog.require('ngeo.attributesDirective');
+/** @suppress {extraRequire} */
+goog.require('ngeo.btnDirective');
+/** @suppress {extraRequire} */
+goog.require('ngeo.createfeatureDirective');
+goog.require('ngeo.EventHelper');
+goog.require('ngeo.ToolActivate');
+goog.require('ngeo.ToolActivateMgr');
+goog.require('ol.format.GeoJSON');
+
+
+/**
+ * Directive used to insert, modify and delete features from a single layer.
+ * It allows you can modify the geometry of the feature in addition to its
+ * attributes.
+ *
+ * In order to modify or delete a feature, you must click on the map at the
+ * location of the feature to select it first.
+ *
+ * In order to create a new feature, you use the "Draw" button and digitalize
+ * the feature on the map.
+ *
+ * Example:
+ *
+ *     <gmf-editfeature
+ *         gmf-editfeature-layer="::ctrl.layer">
+ *         gmf-editfeature-map="::ctrl.map"
+ *         gmf-editfeature-vector="::ctrl.vectorLayer">
+ *     </gmf-editfeature>
+ *
+ * @htmlAttribute {ol.Map} gmf-editfeature-map The map.
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfEditfeature
+ */
+gmf.editfeatureDirective = function() {
+  return {
+    controller: 'GmfEditfeatureController',
+    scope: {
+      'layer': '=gmfEditfeatureLayer',
+      'map': '<gmfEditfeatureMap',
+      'vectorLayer': '<gmfEditfeatureVector'
+    },
+    bindToController: true,
+    controllerAs: 'efCtrl',
+    templateUrl: gmf.baseTemplateUrl + '/editfeature.html'
+  };
+};
+
+gmf.module.directive(
+  'gmfEditfeature', gmf.editfeatureDirective);
+
+
+/**
+ * @param {!angular.Scope} $scope Angular scope.
+ * @param {angular.$timeout} $timeout Angular timeout service.
+ * @param {gmf.EditFeature} gmfEditFeature Gmf edit feature service.
+ * @param {gmf.XSDAttributes} gmfXSDAttributes The gmf XSDAttributes service.
+ * @param {ngeo.EventHelper} ngeoEventHelper Ngeo Event Helper.
+ * @param {ngeo.ToolActivateMgr} ngeoToolActivateMgr Ngeo ToolActivate manager
+ *     service.
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfEditfeatureController
+ */
+gmf.EditfeatureController = function($scope, $timeout, gmfEditFeature,
+    gmfXSDAttributes, ngeoEventHelper, ngeoToolActivateMgr) {
+
+  /**
+   * @type {GmfThemesNode}
+   * @export
+   */
+  this.layer;
```

Comment:
We don't need to export it.  It is the property that is assigned automatically by the directive `scope` and `bindTocontroller` (see above). However, if we don't, the Closure compiler would rename it and therefore the assignement would not work properly.  The other way to avoid having this marked as  `@export` would be to do something like this:

```
this.layer_ = this['layer'];
```

and mark it as `@private`.  But then again, the property would be set twice, which is pretty useless.  This is something I have already explained to Fred a couple weeks ago and he agreed with leaving this "as-is".

What do you think ?